lib/Logger.js

```diff
@@ -32,20 +36,28 @@ class Logger {
      *                                      the 'level' field is not accounted
      *                                      for)
      *
-     * @param {string} name  - The name of the Logger. It can be found later on
-     *                         in the log entries.
      *
      * @see [Bunyan's documentation]{@link
      * https://github.com/trentm/node-bunyan/blob/master/README.md#streams} for
      * a more detailed description of the streams array configuration.
      *
      * @returns {undefined}
      */
-    constructor(config, name) {
-        this.config = config;
+    constructor(name, config) {
+        this.config = new Config();
+        if (config) {
+            this.config.update(config);
+        }
         this.name = name;
     }
 
+    setLevel(levelName) {
```

Comment:
Do you want to enforce keeping those in the Logger class ? I could understand that for backwards compatibility if anyone was using it; but afaik, no one is using that in any S3-related repo. As such, we probably want to remove it, and keep the config handling into the `API` class.